src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/SdkDirectoryWorkloadManifestProvider.cs

```diff
@@ -102,23 +107,28 @@ internal SdkDirectoryWorkloadManifestProvider(string sdkRootPath, string sdkVers
 
             _manifestRoots ??= Array.Empty<string>();
 
+            RefreshWorkloadManifests();
+        }
+
+        public void RefreshWorkloadManifests()
+        {
             var availableWorkloadSets = GetAvailableWorkloadSets();
 
-            if (workloadSetVersion != null)
+            if (_workloadSetVersionFromConstructor != null)
             {
-                if (!availableWorkloadSets.TryGetValue(workloadSetVersion, out _workloadSet))
+                if (!availableWorkloadSets.TryGetValue(_workloadSetVersionFromConstructor, out _workloadSet))
                 {
-                    throw new FileNotFoundException(string.Format(Strings.WorkloadVersionNotFound, workloadSetVersion));
+                    throw new FileNotFoundException(string.Format(Strings.WorkloadVersionNotFound, _workloadSetVersionFromConstructor));
                 }
             }
             else
             {
-                string? globalJsonWorkloadSetVersion = GlobalJsonReader.GetWorkloadVersionFromGlobalJson(globalJsonPath);
+                string? globalJsonWorkloadSetVersion = GlobalJsonReader.GetWorkloadVersionFromGlobalJson(_globalJsonPathFromConstructor);
                 if (globalJsonWorkloadSetVersion != null)
```

Comment:
Should this be `NullOrWhitespace` instead? Is it possible we get back an empty string in the global.json for the workloadset version since it's treating it as a raw string and not an actual version it's trying to parse.

Reply:
Someone could probably put an empty string in the global.json.  But this isn't changing in this PR, so I would prefer to leave it as it is in a servicing change.